oarepo_ui/theme/assets/semantic-ui/js/oarepo_ui/forms/components/EDTFDatePickerField/EDTFDatePickerField.jsx

```diff
@@ -0,0 +1,223 @@
+import React, { useState } from "react";
+import DatePicker from "react-datepicker";
+import "react-datepicker/dist/react-datepicker.css";
+import { useField, useFormikContext } from "formik";
+import { DatePickerHeader } from "./DatePickerHeader";
+import PropTypes from "prop-types";
+import { FieldLabel } from "react-invenio-forms";
+import { Form } from "semantic-ui-react";
+import { i18next } from "@translations/oarepo_ui/i18next";
+
+const edtfDateFormatOptions = [
+  { value: "yyyy", text: i18next.t("YYYY") },
+  { value: "yyyy-MM", text: i18next.t("YYYY-MM") },
+  { value: "yyyy-MM-dd", text: i18next.t("YYYY-MM-DD") },
+];
+
+const useInitialDateFormat = (fieldValue) => {
```

Comment:
Aren't there any tools to detect date format from string, or can't we do a simple regex match here?

Reply:
@mirekys Ahoj, the value coming from BE, is passed through native JS Date class because this datepicker works with Date objects and not strings. No mater in which shape the string arrives, if it can be handled by Date class, it will be displayed in the input normally, but it will fail validation if you try to save the work. 

![image](https://github.com/oarepo/oarepo-ui/assets/34883098/688917cf-4f09-412d-b92f-e55e187d61d6)

I have included a try catch block that console.logs an error in case the string coming from the BE has invalid format. 

Though if this does happen to be the case, anyway, I think the only way for user to actually save or do anything with this record would be to choose the proper date via the datepicker.